src/ln/channelmonitor.rs

```diff
@@ -1186,7 +1186,90 @@ impl ChannelMonitor {
 				}
 			}
 		} else {
-			//TODO: For each input check if its in our remote_commitment_txn_on_chain map!
+			let remote_commitment_txn_on_chain = self.remote_commitment_txn_on_chain.lock().unwrap();
+
+			let mut total_value = 0;
+			let mut inputs = Vec::new();
+			let mut key_materials = Vec::new();
+
+			for (txid, commitment_number) in remote_commitment_txn_on_chain.iter() {
+				if tx.input[0].previous_output.txid == *txid { // We're a HTLC-Timeout or HTLC-Success spending a remote commitment tx
+					let secret = self.get_secret(*commitment_number).unwrap();
+					let per_commitment_key = ignore_error!(SecretKey::from_slice(&self.secp_ctx, &secret));
+					let revocation_pubkey = match self.key_storage {
+						KeyStorage::PrivMode { ref revocation_base_key, .. } => {
+							let per_commitment_point = PublicKey::from_secret_key(&self.secp_ctx, &per_commitment_key);
+							ignore_error!(chan_utils::derive_public_revocation_key(&self.secp_ctx, &per_commitment_point, &PublicKey::from_secret_key(&self.secp_ctx, &revocation_base_key)))
+						},
+						KeyStorage::SigsMode { ref revocation_base_key, .. } => {
+							let per_commitment_point = PublicKey::from_secret_key(&self.secp_ctx, &per_commitment_key);
+							ignore_error!(chan_utils::derive_public_revocation_key(&self.secp_ctx, &per_commitment_point, &revocation_base_key))
+						},
+					};
+					let delayed_key = ignore_error!(chan_utils::derive_public_key(&self.secp_ctx, &PublicKey::from_secret_key(&self.secp_ctx, &per_commitment_key), &self.delayed_payment_base_key));
+					let redeemscript = chan_utils::get_revokeable_redeemscript(&revocation_pubkey, self.our_to_self_delay, &delayed_key);
+					let revokeable_p2wsh = redeemscript.to_v0_p2wsh();
+
+
+					if tx.output[0].script_pubkey == revokeable_p2wsh { //HTLC transactions have one txin, one txout
+						inputs.push(TxIn {
```

Comment:
You should try to honor the CLTV_SHARED_CLAIM_BUFFER here - splitting out inputs into separate transactions if they expire too soon.

Reply:
Hmmmm in case of HTLC-Success/Timeout transactions aren't we more affected by CSV delay ? If so I think we want to follow the same delay buffer as for a CLTV but we maybe need to update comment of CLTV_SHARED_CLAIM_BUFFER. And here, if it's HTLC-Success/Timemout transactions, I think we can only have one input so no case of multi-inputs to separate ?